samcli/lib/init/template_modifiers/xray_tracing_template_modifier.py

```diff
@@ -29,27 +32,42 @@ def _add_new_field_to_template(self):
             function_section_position = self._section_position(self.FUNCTION, global_section_position)
 
             if function_section_position >= 0:
-                field_positon = self._field_position(function_section_position, self.FIELD_NAME)
-                if field_positon >= 0:
-                    self.template[field_positon] = self.TRACING
-                    return
+                field_positon_function = self._field_position(
+                    function_section_position, self.FIELD_NAME_FUNCTION_TRACING
+                )
+                if field_positon_function >= 0:
+                    self.template[field_positon_function] = self.TRACING_FUNCTION
 
-                new_fields = [self.TRACING]
-                _section_position = function_section_position
+                else:
+                    self.template = self._add_fields_to_section(function_section_position, [self.TRACING_FUNCTION])
 
             else:
-                new_fields = [self.FUNCTION, self.TRACING]
-                _section_position = global_section_position + 1
+                self.template = self._add_fields_to_section(
+                    global_section_position, [self.FUNCTION, self.TRACING_FUNCTION]
+                )
```

Comment:
I wonder if we should move individual sources into their own functions?

Reply:
@mndeveci Thank you for your review!

There are several patterns, such as adding the global section itself, adding only the API section or Function section, and adding only the field without affecting the existing fields.

We can consider separating them into individual Functions, but this can be a complex implementation.

Reply:
I've pushed a change to your PR, please let me know if you are good with it.

Reply:
@mndeveci 
thank you! I have confirmed the update! It has improved a lot.